build-tools/packages/build-cli/src/library/package.ts

```diff
@@ -650,6 +650,26 @@ export async function setVersion(
 	}
 }
 
+/**
+ * Extracts the dependencies record from a package.json file based on the dependency group.
+ */
+function getDependenciesRecord(
+	packageJson: Package["packageJson"],
+	depClass: "prod" | "dev" | "peer",
+): Partial<Record<string, string>> | undefined {
```

Comment:
What does `Partial<>` here give us over simply the Record?

Reply:
I am not sure. It could be away to enforce checking like `exactOptionalPropertyTypes`.
This comes from type-fest:
`export type Dependency = Partial<Record<string, string>>;`
Fixed to use the types from the properties directly.